### PR TITLE
[Bug] Filtering the nodes based on label for ocs inventory

### PR DESCRIPTION
**Pull request by cloudbehl**
Signed-off-by: Ankush Behl <cloudbehl@gmail.com>

### CODE REVIEW

**Comment by cloudbehl** [comment]
@afreen23 @rawagner please review.

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+  );
+
+export const getFilteredPVsData = (pvsData: K8sResourceKind[]): K8sResourceKind[] =>
+  pvsData.filter((pv: K8sResourceKind) =>
+    cephStorageProvisioners.includes(
+      _.get(pv, 'metadata.annotations["pv.kubernetes.io/provisioned-by"]'),
+    ),
+  );
+
+export const getFilteredNodesData = (NodesData: K8sResourceKind[]): K8sResourceKind[] =>
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/selectors/index.ts]:
```suggestion
export const getFilteredNodesData = (nodesData: K8sResourceKind[]): K8sResourceKind[] =>
```

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+  NodesData.filter((node: K8sResourceKind) =>
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/selectors/index.ts]: nit: you dont need to specify again that node is `K8sResourceKind` - it is derived by TS automatically (applies to all `getFiltered*` functions)

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+    Object.keys(_.get(node, 'metadata.labels')).includes(cephStorageLabel),
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/selectors/index.ts]: `Object.keys(_.get(node, 'metadata.labels'))` fails if node has no labels

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
 import { Alert } from '@console/internal/components/monitoring';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+
+const cephStorageProvisioners = ['ceph.rook.io/block', 'cephfs.csi.ceph.com', 'rbd.csi.ceph.com'];
+const cephStorageLabel = 'cluster.ocs.openshift.io/openshift-storage';
 
 export const filterCephAlerts = (alerts: Alert[]): Alert[] =>
   alerts.filter((alert) => _.get(alert, 'annotations.storage_type') === 'ceph');
+
+export const getFilteredPVCsData = (pvcsData: K8sResourceKind[]): K8sResourceKind[] =>
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/selectors/index.ts]: i'd add default value
```suggestion
export const getFilteredPVCsData = (pvcsData: K8sResourceKind[] = []): K8sResourceKind[] =>
```

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts (hunk 3 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/selectors/index.ts]: loadash `keys` would be safe to use here as it returns an empty array for null/undefined objects

**Comment by cloudbehl** [comment]
@rawagner thanks for the review, made the requested changes.

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts [hunk 5]
```diff
@@ -1,5 +1,28 @@
...
 import { Alert } from '@console/internal/components/monitoring';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+
+const cephStorageProvisioners = ['ceph.rook.io/block', 'cephfs.csi.ceph.com', 'rbd.csi.ceph.com'];
+const cephStorageLabel = 'cluster.ocs.openshift.io/openshift-storage';
 
 export const filterCephAlerts = (alerts: Alert[]): Alert[] =>
   alerts.filter((alert) => _.get(alert, 'annotations.storage_type') === 'ceph');
+
+export const getFilteredPVCsData = (pvcsData: K8sResourceKind[] = []): K8sResourceKind[] =>
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/selectors/index.ts]: one more thing, as these functions are generic selectors which are meant be used in other places than dashboard they should have a more descriptive names such as

```suggestion
export const getCephPVCs = (pvcs: K8sResourceKind[] = []): K8sResourceKind[] =>
```
`export const getCephPVs = (pvs: K8sResourceKind[] = []): K8sResourceKind[] =>`
`export const getCephNodes = (nodes: K8sResourceKind[] = []): K8sResourceKind[] =>`

**Comment by cloudbehl** [comment]
@rawagner updated the PR.

**Pull request merged**